parquet-pig/src/main/java/org/apache/parquet/pig/TupleWriteSupport.java

```diff
@@ -172,7 +172,7 @@ private void writeValue(Type type, FieldSchema pigType, Tuple t, int i) {
           } else {
             throw new UnsupportedOperationException("can not convert from " + DataType.findTypeName(pigType.type) + " to BINARY ");
           }
-          recordConsumer.addBinary(Binary.fromByteArray(bytes));
+          recordConsumer.addBinary(Binary.fromConstantByteArray(bytes));
```

Comment:
I think this is correct for the String case, but how do we know that the DataByteArray case will not be a reused object? You could always change bytes to a Binary and wrap the byte array inside the if blocks.